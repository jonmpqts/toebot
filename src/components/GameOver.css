.gameover {
    background: rgba(255, 255, 255, .8);
    bottom: 5px;
    color: #3d4250;
    left: 5px;
    padding-top: 110px;
    position: absolute;
    right: 5px;
    text-align: center;
    top: 5px;
    width: 450px;
    z-index: 11;
}

.gameover h2 {
    font-size: 40px;
    font-weight: 300;
}