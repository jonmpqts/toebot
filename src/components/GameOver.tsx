import * as React from 'react';
import './GameOver.css';

interface Props {
    winner: 'X' | 'O' | 'XO';
}

const GameOver = ({ winner }: Props) => {
    const text = winner === 'XO'
        ? 'It is a tie!'
        : 'Player ' + (winner === 'X' ? '1' : '2') + ' wins!';

    return (
        <div className="gameover">
            <h2>{text}</h2>
        </div>
    );
};

export default GameOver;
