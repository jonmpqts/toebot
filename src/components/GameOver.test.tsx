import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import GameOver from './GameOver';

test('first player wins', () => {
    const component = renderer.create(
        <GameOver
            winner="X"
            onRestartClick={() => { return; }}
        />
    ); 

    expect(component.toJSON()).toMatchSnapshot();
});

test('second player wins', () => {
    const component = renderer.create(
        <GameOver
            winner="O"
            onRestartClick={() => { return; }}
        />
    ); 

    expect(component.toJSON()).toMatchSnapshot();
});

test('draw', () => {
    const component = renderer.create(
        <GameOver
            winner="XO"
            onRestartClick={() => { return; }}
        />
    ); 

    expect(component.toJSON()).toMatchSnapshot();
});

test('responds to restart click', () => {
    let clicked = false;

    const wrapper = mount(
        <GameOver
            winner="X"
            onRestartClick={() => { clicked = true; }}
        />
    );

    wrapper.find('button').simulate('click');
    expect(clicked).toBeTruthy();
});
