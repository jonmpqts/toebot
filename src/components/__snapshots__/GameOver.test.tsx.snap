// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`draw 1`] = `
<div
  className="gameover"
>
  <h2>
    It is a tie!
  </h2>
  <button
    onClick={[Function]}
  >
    Restart
  </button>
</div>
`;

exports[`first player wins 1`] = `
<div
  className="gameover"
>
  <h2>
    Player 1 wins!
  </h2>
  <button
    onClick={[Function]}
  >
    Restart
  </button>
</div>
`;

exports[`second player wins 1`] = `
<div
  className="gameover"
>
  <h2>
    Player 2 wins!
  </h2>
  <button
    onClick={[Function]}
  >
    Restart
  </button>
</div>
`;
